interfaces/api/v1/core/settings.py

```
@@ -0,0 +1,153 @@
+    "claude-sonnet-4-6",
+    "claude-3-5-sonnet-20241022",
+    "claude-3-opus-20240229",
+    "claude-3-5-haiku-20241022",
+    "claude-3-haiku-20240307",
+]
+
+
+@router.post("/fetch-models")
+async def fetch_models(body: FetchModelsRequest):
+    if body.provider == "anthropic":
+        return body.base_url and await _fetch_openai_models(body.api_key, body.base_url) or _ANTHROPIC_MODELS
```

Ruff (0.15.10)

[warning] Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear

Parenthesize the `and` subexpression

(RUF021)